src/components/PopupWithDelete.js

```diff
@@ -0,0 +1,33 @@
+import Popup from "./Popup";
+
+export default class PopupWithDelete extends Popup {
+  constructor(popupSelector, loadingButtonText) {
+    super(popupSelector);
+    this._buttonText = this._submitButton.textContent;
+    this._loadingButtonText = loadingButtonText;
+  }
+
+  showLoading() {
+    this._submitButton.textContent = this._loadingButtonText;
+  }
+
+  hideLoading() {
+    this._submitButton.textContent = this._buttonText;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+
+    this._modalContainer.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleDeleteFormSubmit().then(() => {
```

Comment:
Needs correction: this class should not know that `this._handleDeleteFormSubmit` is a promise

Reply:
Is there a particular reason for this? I don't remember learning about that or seeing that in the code checklist.

Reply:
The class should not know anything about how the handler is implemented. It can return a promise or it can return anything else.